subprojects/execution/src/test/groovy/org/gradle/internal/execution/impl/steps/ExecutionTest.groovy

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.execution.impl.steps
+
+import com.google.common.collect.ImmutableSortedMap
+import org.gradle.api.BuildCancelledException
+import org.gradle.api.file.FileCollection
+import org.gradle.caching.internal.origin.OriginMetadata
+import org.gradle.initialization.DefaultBuildCancellationToken
+import org.gradle.internal.execution.CacheHandler
+import org.gradle.internal.execution.ExecutionException
+import org.gradle.internal.execution.ExecutionOutcome
+import org.gradle.internal.execution.OutputChangeListener
+import org.gradle.internal.execution.UnitOfWork
+import org.gradle.internal.execution.history.changes.ExecutionStateChanges
+import org.gradle.internal.fingerprint.CurrentFileCollectionFingerprint
+import spock.lang.Specification
+
+import java.time.Duration
+import java.util.function.BooleanSupplier
+
+class ExecutionTest extends Specification {
+
+    def outputChangeListener = Mock(OutputChangeListener)
+    def cancellationToken = new DefaultBuildCancellationToken()
+    def executionStep = new CatchExceptionStep<Context>(
```

Comment:
I don't feel very strongly, but it would be nice to find a better name for this guy.